autotest/pyscripts/test_gdal2tiles.py

```diff
@@ -89,23 +100,22 @@ def _verify_raster_band_checksums(filename, expected_cs=[]):
 
 
 @pytest.mark.require_driver("PNG")
-def test_gdal2tiles_py_simple(script_path):
+def test_gdal2tiles_py_simple(script_path, tmp_path):
+
+    input_tif = str(tmp_path / "out_gdal2tiles_smallworld.tif")
+    output_dir = input_tif.strip(".tif")
 
     shutil.copy(
         test_py_scripts.get_data_path("gdrivers") + "small_world.tif",
-        "tmp/out_gdal2tiles_smallworld.tif",
+        input_tif,
     )
 
-    os.chdir("tmp")
     test_py_scripts.run_py_script(
-        script_path, "gdal2tiles", "-q out_gdal2tiles_smallworld.tif"
+        script_path, "gdal2tiles", f"-q {input_tif} {output_dir}"
```

Comment:
this no longer tests a non specified output dir. do we have another test where output_dir is not specified?

Reply:
I must have thought (incorrectly, I guess) that I couldn't call `os.chdir` when running tests in parallel. I reverted this to use chdir and added a comment explaining the purpose.